### ISSUE TITLE
How does one build a local version of the extension?

### ISSUE THREAD

**Issue by vsaulys** [KxSystems/kx-vscode#258]
There is no documentation on this on the project page, and there appears to be an issue when following standard build steps (npm install, npm run package). Run package returns a list of errors about missing dependancies for esbuild:

(base) vssaulys@f84d898f5e35 kx-vscode % npm run package

> kdb@1.3.0 package
> npx vsce package

Executing prepublish script 'npm run vscode:prepublish'...

> kdb@1.3.0 vscode:prepublish
> npm run -S esbuild-base -- --minify --keep-names

> kdb@1.3.0 esbuild-base
> rimraf out && node ./esbuild.js "--minify" "--keep-names"

extension build complete
server build complete
build complete
 ERROR  Command failed: npm list --production --parseable --depth=99999 --loglevel=error
npm ERR! code ELSPROBLEMS
npm ERR! missing: @esbuild/aix-ppc64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/android-arm@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/android-arm64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/android-x64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/darwin-arm64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/freebsd-arm64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/freebsd-x64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-arm@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-arm64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-ia32@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-loong64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-mips64el@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-ppc64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-riscv64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-s390x@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/linux-x64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/netbsd-x64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/openbsd-x64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/sunos-x64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/win32-arm64@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/win32-ia32@0.19.12, required by esbuild@0.19.12
npm ERR! missing: @esbuild/win32-x64@0.19.12, required by esbuild@0.19.12

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/vssaulys/.npm/_logs/2024-03-06T17_34_18_068Z-debug.log

**Comment by lysliu** [comment]
I build it out on macbook pro with M1 processor
```
git clone git@github.com:KxSystems/kx-vscode.git
cd kx-vscode
npm run package
brew search npm
brew install npm
brew update
npm run package
npm install -g npm@10.5.0
npm run package
npm install -g rimraf\n
npm run package
npm install esbuild\n
```
after clone the code and install dependencies, get it built out
```
➜  kx-vscode git:(main) ✗ npm run package

> kdb@1.3.0 package
> npx vsce package

(node:60767) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Executing prepublish script 'npm run vscode:prepublish'...

> kdb@1.3.0 vscode:prepublish
> npm run -S esbuild-base -- --minify --keep-names


> kdb@1.3.0 esbuild-base
> rimraf out && node ./esbuild.js --minify --keep-names

extension build complete
server build complete
build complete
 DONE  Packaged: /Users/liuyansong/kx-vscode/kdb-1.3.0.vsix (84 files, 2.86MB)
```

**Comment by nfarrell-kx** [comment]
Was this your first time building the project, could you try deleting `node_modules`?
`npm install` and `npm run package` should work as build steps

**Comment by vsaulys** [comment]
This is my first project, and for the life of me could not figure things out.

Last issue I ran into was rpm install was simply having, I then just rebooted my machine (because that is the default check? Did you turn it on and off again?)

Anyway, rebooting, and rpm build package just WORKED? On my Mac (M2, Sonoma).

Was able to also build with what I think the fix (src/validators/kdbValidator.ts), just increase the string length checks to 512 from 64.

Also modified the tests to check a longer invalid string (513 characters), change was like this in test/suit/validators.test.ts):

  it("Should return fail for server password that is outside the size limits", () => {
    const result = validateServerPassword(
      //"ttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttt"
      "t".repeat(513),
    );
    assert.strictEqual(
      result,
      "Input value must be between 1 and 512 alphanumeric characters in length.",
      "Input was outside the size limits.",
    );
  });



I’ve attached the modified files, btw, “rpm run test” is failing now, not clear why?

(base) ***@***.*** kx-vscode % npm run test

> ***@***.*** test
> tsc -p ./test && node ./out/test/runTest.js

error TS2688: Cannot find type definition file for 'sinonjs__fake-timers'.
  The file is in the program because:
    Entry point for implicit type library 'sinonjs__fake-timers'


Found 1 error.


Also attaching the modified files, it was an easy fix.

￼￼

Vincent Saulys
***@***.***

**Comment by vsaulys** [comment]
I have found the core issue.

When debugging TLS connections in q, I had configured and set local SSL certs, they apparently where the root cause to my networking related issues.

Thought I would pass this along, it was my having set these:
SSL_CERT_FILE
SSL_KEY_FILE

I have since moved to using the KX named version of these (e.g. KX_ SSL_CERT_FILE) which admittedly I should have done in the first place.

Vincent Saulys
***@***.***
